flamingpy/codes/graphs/stabilizer_graph.py

```diff
@@ -333,25 +333,32 @@ def edges(self):
         return self.graph.edges()
 
     def shortest_paths_without_high_low(self, source, code):
+        self.assign_weights(code)
         subgraph = self.graph.subgraph(
             self.stabilizers + self.low_bound_points + self.high_bound_points
         )
         return nx_shortest_paths_from(subgraph, source, code)
 
     def shortest_paths_from_high(self, code):
+        self.assign_weights(code)
         return nx_shortest_paths_from(self.graph, "high", code)
 
     def shortest_paths_from_low(self, code):
+        self.assign_weights(code)
```

Comment:
Let us supposed the code has open boundaries. In this case, `MatchinGraph.with_edges_from` will call all of these functions, which means `assign_weights` will be called 3 times. Should `with_edges_from` call `assign_weights`?

Reply:
But, I agree it is annoying, but it was already like in previous version. If you care about speed, you should use the retworkx or lemon backend anyway.

Reply:
I see. But how about having this in `MatchingGraph.with_edges_from`:

```
if isinstance(stab_graph, NxStabilizerGraph):
    stab_graph.assign_weights(code)
```

and then remove `assign_weights` from the shortest_paths functions.

Reply:
We can do that, but I would prefer to contain implementation details into the Networkx class.

Reply:
Also, if we remove the assign_weights call from the shortest path functions, they can't be called without a manual call to assign_weights before. Which is a bit annoying.

Reply:
> We can do that, but I would prefer to contain implementation details into the Networkx class.

I generally agree with this policy, but here we have a choice between making the abstract class slightly less general and having two redundant function calls. I may be leaning towards the latter unless there is an alternative solution.

> Also, if we remove the assign_weights call from the shortest path functions, they can't be called without a manual call to assign_weights before. Which is a bit annoying.

This is true, but is it annoying? Technically it is meaningless to compute shortest paths before the weights have been assigned.

Reply:
For the retworkx backend, you don't need to pre-assign the weights. They are computed on the fly when needed. Thus, the same function would have different behavior depending on the backend.

Reply:
Also, we already called the decoder.assign_weights function to assign weights to the code nodes. It would be weird to then have to call code.stab_graph.assign_weights. 

That being said, it is maybe a good idea to also assign the weights to the stabilizer graph when we call decoder.assign_weights.

Reply:
Good point - ignoring some changes that we have planned for the future, what do you say to making the change in this PR? Moving the weight assignment to decoder.assign_weights if the code backend is 'networkx'? For the moment there will be a bit of a redundancy in that weights will both be assigned to the EGraph and the stabilizer graph. However, this is nice for debugging/visualization purposes, and we will not need to iterate again through the EGraph when assigning weights in the stabilizer graph.